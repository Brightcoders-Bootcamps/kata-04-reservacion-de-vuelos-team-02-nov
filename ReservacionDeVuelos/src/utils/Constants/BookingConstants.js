const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next'; 
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';

export const BookingConstants = {
    TitleOrigin,
    TitleDestination,
    TextInput,
    BtnText, 
    SelectDate,
    Passengers,
};