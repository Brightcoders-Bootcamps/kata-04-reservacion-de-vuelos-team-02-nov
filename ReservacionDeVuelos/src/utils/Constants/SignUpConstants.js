const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage = 'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
};
