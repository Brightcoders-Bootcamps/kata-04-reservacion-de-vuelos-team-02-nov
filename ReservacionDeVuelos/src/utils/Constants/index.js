export * from './SignUpConstants';
export * from './FlightsListConstants'
