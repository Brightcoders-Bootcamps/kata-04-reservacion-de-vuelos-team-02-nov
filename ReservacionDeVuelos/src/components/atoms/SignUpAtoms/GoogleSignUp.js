import React from 'react';
import { Text, TouchableOpacity, Image} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import GoogleIcon from '../../../assets/images/google.png';
import {SignUpConstants} from '../../../utils/Constants';
const GoogleSignUp = () => {
  return (
    <TouchableOpacity style={styles.btnSignup} testID='GoogleSignUp'>
      <Image
      testID='GoogleIcon'
        source={GoogleIcon}
        style={[styles.googleIcon, {marginLeft: -75}]}
      />
      <Text style={[styles.btnSignupText, {marginLeft: 50}]} testID='btnTitle'>
        {SignUpConstants.GoogleSignUpButton}
      </Text>
    </TouchableOpacity>
  );
};

export default GoogleSignUp;
