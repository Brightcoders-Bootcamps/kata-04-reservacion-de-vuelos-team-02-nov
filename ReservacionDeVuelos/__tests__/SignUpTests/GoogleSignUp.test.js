import React from 'react';
import GoogleSignUp from '../../src/components/atoms/SignUpAtoms/GoogleSignUp';
import {shallow} from 'enzyme';
import {SignUpConstants} from '../../src/utils/Constants';
import GoogleIcon from '../../src/assets/images/google.png';
import {store} from '../../redux/store';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<GoogleSignUp store={store} />);
});

describe('renders correclty', () => {
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('component exists', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'GoogleSignUp',
    );
    expect(component.exists()).toBeTruthy();
  });
  test('Icon img is displayed with the correct source', () => {
    const img = wrapper.findWhere(
      (node) => node.prop('testID') === 'GoogleIcon',
    );
    expect(img.props().source).toEqual(GoogleIcon);
  });
  test('btn title renders correctly', () => {
    const btnTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );

    expect(btnTitle.props().children).toEqual(
      SignUpConstants.GoogleSignUpButton,
    );
  });
});
