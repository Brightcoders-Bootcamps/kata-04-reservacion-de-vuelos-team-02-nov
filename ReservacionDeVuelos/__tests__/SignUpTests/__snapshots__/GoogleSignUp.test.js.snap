// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty snapshot 1`] = `
<GoogleSignUp
  signupGoogleFunction={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
